### PR TITLE
Add Health Status to workload API

### PR DESCRIPTION
**Pull request by howardjohn**
ztunnel side https://github.com/istio/ztunnel/pull/374

### CODE REVIEW

**Comment by bleggett** [comment]
Have some concerns about letting pod states leak into `ztunnel` mentioned here: https://github.com/istio/ztunnel/pull/374#pullrequestreview-1283577090

(if there's a valid case I'm ignoring, then nevermind)

File: pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go [hunk 1]
```diff
@@ -354,103 +354,6 @@ func TestAmbientIndex(t *testing.T) {
 		workloadapi.Protocol_HTTP)
 }
 
-func TestPodLifecycleWorkloadGates(t *testing.T) {
```
Comment (bleggett) [inline pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go:357]: While the underlying logic is changed, I would like to retain explicit unit tests verifying various combinations of lifecycle state changes produce the expected workload entry.

Comment (howardjohn) [reply pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go:357]: Good call, added back

**Review by bleggett — APPROVED** [review]
LGTM. Apologies for the confusion.

**Comment by linsun** [comment]
Thanks John! This makes sense, I got confused earlier thought you meant adding all kind of status.  LGTM

**Review by dhawton — APPROVED** [review]

**Pull request merged**